Development-Guide.md

```diff
@@ -1,18 +1,18 @@
-# Table of Contents
+<h1 id="table-of-contents">Table of Contents</h1>
 
 - [C++](#c)
 - [Lua](#lua)
 - [SQL](#sql)
 - [Python](#python)
 - [Git](Development-Guide-Git)
 
-## Contributing
+<h2 id="contributing">Contributing</h2>
 
 It is assumed you've read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md).
 
-## C++
+<h2 id="c">C++</h2>
 
-### Naming & Misc
+<h3 id="c-naming-misc">Naming & Misc</h3>
```

Comment:
what's the goal here? Maybe I'm dense I'm not understanding.

Reply:
It allows for clearer linking. For instance using the normal markdown anchor links (which are also still available) it's unclear what [#naming--misc](https://github.com/LandSandBoat/server/wiki/Development-Guide#naming--misc), [#naming-and-misc](https://github.com/LandSandBoat/server/wiki/Development-Guide#naming-and-misc), and [#naming-and-misc-1](https://github.com/LandSandBoat/server/wiki/Development-Guide#naming-and-misc-1) refer to. This can become `#c-naming-misc`, `#lua-naming-misc`, and `#python-naming-misc` without having to actually change the header text, and also allows you to edit header text without having to go update links elsewhere.

Reply:
So we're working around a parsing bug in githubs Gollum implementation then - spaces are allowed in the spec, you shouldn't have to convert anything and pages inside the wiki (not external website) should look like 
```md
[[page name#anchor name]]
```
or
```md
[[optional non name text|page name#anchor name]]
```

I'll accept\* html'd headings on one condition: we plan to eventually clean up all these cases where we mistakenly have used external style links to refer to internal locations.


###### \*I'm admittedly markdown chauvinist

Reply:
It's not about internal wiki links, but external links linking to a specific part of the page, but in either case you rely on the actual text of the header. Nothing to do with spaces. This just adds an id as an alternative name to link to. If I want to link specifically to the SQL naming section of the page, I can use:

`https://github.com/LandSandBoat/server/wiki/Development-Guide#naming-and-misc-1`
or
`https://github.com/LandSandBoat/server/wiki/Development-Guide#sql-naming-and-misc`

If at some point the text of that heading is changed, the link to `#sql-naming-and-misc` will still work as long as the id isn't changed.

Maybe I can visualize.
This is how we see the page:
```
H1
    H2
        H3
    H2
```
This is how we link to the page:
```
#H1
#H2
#H3
#H2-1 (if name collision)
```
but it's useful to have that H1 header attached because of the way the information on the page is organized, otherwise it's ambiguous without being unnecessarily descriptive in the heading text.

Reply:
you misunderstand what I am saying: I'm offering a trade. I'll not gripe about html in my pure beautiful markdown if we _also_ correct the link types at a later date (we have a bunch of internal pages being linked as external)

Reply:
(I'm also trying to be funny, calling md "beautiful")

Reply:
Rest of wiki will def get a cleaning up. This one just gets special attention cause it's important styleguide.